aspen/request_processor/algorithm.py

```diff
@@ -70,6 +70,11 @@ def load_resource_from_filesystem(request_processor, dispatch_result):
     return {'resource': resources.get(request_processor, dispatch_result.match)}
 
 
-def render_resource(state, request_processor, resource):
-    state.setdefault('output', output.Output(charset=request_processor.charset_dynamic))
+def render_resource(state, resource):
     return {'output': resource.render(state)}
+
+
+def encode_output(output, request_processor):
+    if isinstance(output.body, unicode):
+        output.charset = request_processor.charset_dynamic
```

Comment:
Well, now that you jog my memory, I believe the point of the `charset_static` knob is that if you send a `text/plain` file without specifying a charset, then browsers will assume it's `latin-1`, which seems like a bad default in the age of `utf-8`. So `charset_static` is a way to override this web-wide default for Aspen applications.

`charset_dynamic` is used in the `Content-Type` header (as with `charset_static`), _and_ to encode the body (for static resources the body can be thought of as already having been encoded).

Maybe something like `encode_dynamic_resources_as` and `assume_static_resources_are_encoded_as`?